### DISCUSSION TITLE
Track messages changes of a user with Delta with Token

### DISCUSSION THREAD

**Discussion by maxklav** [microsoftgraph/msgraph-sdk-go#562] [Q&A] [answered]
`DeltaWithToken()` is not provided with all request builders

```go
var client *msgraphsdk.GraphServiceClient
// ...

// Works with a user's onedrive drive items
- = Drives().ByDriveId(driveID).Items().ByDriveItemId("root").Delta()
- = Drives().ByDriveId(driveID).Items().ByDriveItemId("root").DeltaWithToken("...")

// It does not work with a user's mail folder messages
- = client.Users().ByUserId(userID).Messages().Delta()
- = client.Users().ByUserId(userID).Messages().DeltaWithToken("...") // This does NOT work

// It also does not work if we try to go by mail folders:
- = client.Users().ByUserId(userID).MailFolders().Delta()
- = Users().ByUserId(userID).MailFolders().DeltaWithToken("...") // This does NOT work
- = client.Users().ByUserId(userID).MailFolders().ByMailFolderId(mailFolderID).Messages().Delta()
- = client.Users().ByUserId(userID).MailFolders().ByMailFolderId(mailFolderID).Messages().DeltaWithToken("...")  // This does NOT work
```

We would like to list **all** messages of all mail folders of a user in the first run of our tool and list created/updated/deleted messages on the other runs.

How can we do this ?

**Comment by baywet** [comment, marked as answer]
Please have a look at [that discussion](https://github.com/microsoftgraph/msgraph-sdk-go/discussions/543)

Also the delta with token methods should not exist, please avoid using them [here is the rationale](https://github.com/microsoftgraph/msgraph-metadata/issues/354)

Let us know if you have additional questions.

**Reply by maxklav** [reply]
Ok we cannot use this, it's not the SDK that is missing it. But I'm still not sure to understand how it can fix my problem:

> We would like to list all messages of all mail folders of a user in the first run of our tool and list created/updated/deleted messages on the other runs
>
> How can we do this ?

**Reply by baywet** [reply]
This is an API limitation today. You can't track changes for entire content of the mailbox in a single call/at a single endpoint. Since this is an API functionality and this repository is focused on the SDK itself (regardless of the API), you should [request or upvote the feature](https://aka.ms/graphfeedback)
